src/main/java/hudson/plugins/mercurial/Cache.java

```diff
@@ -216,6 +216,15 @@ private synchronized ReentrantLock getLockForSlaveNode(String node) {
                         return null;
                     }
                 }
+				
+				// Fix for JENKINS-33255: if the clone of the repository on the master indicates that it requires largefiles, then 
+				// pull them for the slave's cached repository. When the repository is the cloned/updated into the workspace, the 
+				// clone can then bring in all the largefiles without a problem. Largefiles are shared (hardlinked) on a single machine 
+				// so we will not end up with extra copies.
+				if (masterHg.requiresLargefiles(masterCache)) {
+					slaveHg.lfpull(localCache, null, remote);
```

Comment:
This is not quite right; we need to pull largefiles for the revision we are actually going to use in the workspace, not just for '.' (which would be tip, and might be a totally different branch to the one we actually want to use).